### PR TITLE
Openssl branch 1.1. rebased on top of the current master

### PR DESCRIPTION
**Pull request by catiamesquitactw**
- Fixed compilation issues
- Old and new functionality are tested with the library

### CODE REVIEW

File: src/openssl_wrap.cpp [hunk 1]
```diff
@@ -949,5 +1012,116 @@ void _X509_STORE_CTX_set_time(X509_STORE_CTX *ctx, time_t time)
...
+}
+
+void _CRYPTO_malloc_init()
+{
+    return lib::OpenSSLLib::SSL_CRYPTO_malloc_init();
+}
+
+EC_KEY *_EVP_PKEY_get0_EC_KEY(EVP_PKEY *pkey)
+{
+    OpensslCallPtr::callChecked(lib::OpenSSLLib::SSL_EVP_PKEY_get0_EC_KEY,pkey);
```
Comment (fmonteiro93) [inline src/openssl_wrap.cpp]: Missing return statement

File: src/mococrw/asymmetric_encryption.h [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+
+#include <type_traits>
+#include <cstring>
```
Comment (fmonteiro93) [inline src/mococrw/asymmetric_encryption.h]: Move this to asymmetric_encryption.cpp

File: src/mococrw/openssl_lib.h [hunk 3]
```diff
@@ -248,6 +264,44 @@ class OpenSSLLib
...
+    static int SSL_EVP_PKEY_CTX_set_rsa_oaep_label(EVP_PKEY_CTX *ctx, unsigned char *l, int llen) noexcept;
+    static int SSL_EVP_PKEY_CTX_get_rsa_oaep_label(EVP_PKEY_CTX *ctx, unsigned char *l) noexcept;
+
+    static int SSL_RSA_size(const RSA *r) noexcept;
+    static int SSL_EVP_MD_size(const EVP_MD *md) noexcept;
+    static int SSL_EVP_PKEY_CTX_set_rsa_mgf1_md(EVP_PKEY_CTX *ctx, const EVP_MD *md) noexcept;
+    static int SSL_EVP_PKEY_CTX_set_rsa_padding(EVP_PKEY_CTX *ctx, int pad) noexcept;
+
+
+
```
Comment (fmonteiro93) [inline src/mococrw/openssl_lib.h]: Remove blank lines

File: src/openssl_lib.cpp [hunk 4]
```diff
@@ -606,6 +664,112 @@ void OpenSSLLib::SSL_X509_STORE_CTX_set_time(X509_STORE_CTX* ctx, unsigned long
...
+int OpenSSLLib::SSL_EVP_PKEY_CTX_set_rsa_mgf1_md(EVP_PKEY_CTX *ctx, const EVP_MD *md) noexcept
+{
+    return EVP_PKEY_CTX_set_rsa_mgf1_md(ctx, md);
+}
+EC_KEY *OpenSSLLib::SSL_EVP_PKEY_get0_EC_KEY(EVP_PKEY *pkey) noexcept
+{
+    return EVP_PKEY_get0_EC_KEY(pkey);
+}
+
+    }  //::lib
```
Comment (fmonteiro93) [inline src/openssl_lib.cpp]: Remove blank lines and trailing spaces

**Comment by wusto** [comment]
Please squash/fixup the last 3 remaining commits.
Apart from that it looks good to me.

**Comment by wusto** [comment]
Seems like something got broken while merging.
Suggestion:

- [ ] Checkout a1aca11960c640075c7b251d0cab1fae405ab44f (Openssl branch 1.1 rebased on top of the current master)
- [ ] rebase to the top of the origin/master (git fetch --all; git rebase origin master)
- [ ] squash the three pending commits (git rebase -i HEAD~5 --> change "pick" to "s" in the lower two commits
- [ ] Make the commit message nice and sexy
- [ ] Upload your changes

**Pull request merged**